addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/SmappeeService.java

```diff
@@ -0,0 +1,473 @@
+                    return;
+                }
+            }
+
+            throw new InvalidConfigurationException("Could not find a valid servicelotion for "
+                    + config.service_location_name + ", check binding configuration");
+        } catch (JsonSyntaxException pe) {
+            throw new CommunicationException("Failed to parse servicelocation response", pe);
+        }
+    }
+
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    private String getData(String request) throws CommunicationException {
+        String url = "https://app1pub.smappee.net" + request;
+
+        Request getMethod = httpClient.newRequest(url);
+        getMethod.agent("Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
+        getMethod.accept("application/json");
+
+        String accessTokenToInclude = getAccessToken();
+
+        getMethod.header("Authorization", "Bearer " + accessTokenToInclude);
+
+        try {
+            ContentResponse response = getMethod.send();
+            if (response.getStatus() != HttpStatus.OK_200) {
+                throw new CommunicationException("Get data method failed : " + response.getReason());
+            }
+
+            return response.getContentAsString();
+        } catch (InterruptedException e) {
+            throw new CommunicationException("Request aborted", e);
+        } catch (TimeoutException e) {
+            throw new CommunicationException("Timeout error", e);
+        } catch (ExecutionException e) {
+            throw new CommunicationException("Communication error", e.getCause());
+        }
+    }
+
+    private void setData(String request, String jsonContent) throws CommunicationException {
+        String url = "https://app1pub.smappee.net" + request;
+
+        Request postMethod = httpClient.newRequest(url);
+        postMethod.method(HttpMethod.POST);
+        postMethod.agent("Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
+        postMethod.accept("application/json");
+        postMethod.content(new StringContentProvider(jsonContent));
+
+        String accessTokenToInclude = getAccessToken();
+
+        postMethod.header("Authorization", "Bearer " + accessTokenToInclude);
+
+        try {
+            ContentResponse response = postMethod.send();
+            if (response.getStatus() != HttpStatus.OK_200) {
+                throw new CommunicationException("Post data method failed : " + response.getReason());
+            }
+
+            return;
+        } catch (InterruptedException e) {
+            throw new CommunicationException("Request aborted", e);
+        } catch (TimeoutException e) {
+            throw new CommunicationException("Timeout error", e);
+        } catch (ExecutionException e) {
+            throw new CommunicationException("Communication error", e.getCause());
+        }
+    }
+
+    private String getAccessToken() throws CommunicationException {
```

Comment:
As of today you have no other way of doing this, so this method currently makes sense. However, you might want to watch out for https://github.com/eclipse/smarthome/pull/5142